### DISCUSSION TITLE
[bug]: I'm completely unable to be synced with the Lightning network.

### DISCUSSION THREAD

**Discussion by ghost** [lightningnetwork/lnd#8708] [Troubleshooting] [answered]
* version of `lnd` - lnd version 0.17.4-beta commit=v0.17.4-beta
* which operating system (`uname -a` on *Nix) - macOS Sonoma
* version of `btcd`, `bitcoind`, or other backend - Bitcoin Core 26.0
----------------------------------------------------

I'm completely unable to be synced with the Lightning network. When I run lnd on the macOS terminal, I am flooded with error messages like this every few seconds:

<img width="980" alt="lnd" src="https://github.com/lightningnetwork/lnd/assets/155139404/af8c0866-ba1c-4f00-8d6b-e793775ff32d">

When I type "lncli getinfo" in the macOS terminal, this is what I get:

  "synced_to_chain":  true,
    "synced_to_graph":  false.

For info, I'm running a Bitcoin Core pruned node.

I have absolutely no idea what I'm doing wrong here. Perhaps the issue is in my lnd.conf or/and bitcoin.conf file ? Please help.

**Comment by saubyk** [comment]
Hi @drkhero is your node connected with any n/w peers? From the log it appears like a network connectivity issue

What's the output of `lncli listpeers`?

**Reply by ghost** [reply]
Hello @saubyk , here's the output of ```lncli listpeers```:


```
{
    "peers":  [
        {
            "pub_key":  "03043ccfad51514afc8a2cb68994df198d8aa2cdaea15dfc2530fb40e8edd7dc61",
            "address":  "185.150.160.209:4007",
            "bytes_sent":  "12520",
            "bytes_recv":  "1671668",
            "sat_sent":  "0",
            "sat_recv":  "0",
            "inbound":  false,
            "ping_time":  "1880972",
            "sync_type":  "PASSIVE_SYNC",
            "features":  {
                "0":  {
                    "name":  "data-loss-protect",
                    "is_required":  true,
                    "is_known":  true
                },
                "5":  {
                    "name":  "upfront-shutdown-script",
                    "is_required":  false,
                    "is_known":  true
                },
                "7":  {
                    "name":  "gossip-queries",
                    "is_required":  false,
                    "is_known":  true
                },
                "9":  {
                    "name":  "tlv-onion",
                    "is_required":  false,
                    "is_known":  true
                },
                "12":  {
                    "name":  "static-remote-key",
                    "is_required":  true,
                    "is_known":  true
                },
                "14":  {
                    "name":  "payment-addr",
                    "is_required":  true,
                    "is_known":  true
                },
                "17":  {
                    "name":  "multi-path-payments",
                    "is_required":  false,
                    "is_known":  true
                },
                "23":  {
                    "name":  "anchors-zero-fee-htlc-tx",
                    "is_required":  false,
                    "is_known":  true
                },
                "27":  {
                    "name":  "shutdown-any-segwit",
                    "is_required":  false,
                    "is_known":  true
                },
                "31":  {
                    "name":  "amp",
                    "is_required":  false,
                    "is_known":  true
                },
                "45":  {
                    "name":  "explicit-commitment-type",
                    "is_required":  false,
                    "is_known":  true
                },
                "2023":  {
                    "name":  "script-enforced-lease",
                    "is_required":  false,
                    "is_known":  true
                }
            },
            "errors":  [],
            "flap_count":  1,
            "last_flap_ns":  "1714471490112408000",
            "last_ping_payload":  "00a0b521077b0c5673af50568eaad1a4e41f72983b982cc7845b010000000000000000004b2609c7cd78f46a2ab6a0f8536b0d709c44fe803380ca8a193c9bfb6ed64bc7a8bc3066db3103172ac6355e"
        },
        {
            "pub_key":  "0278b03174171c2d2c817035a6996fae51bee28a9c6f394c2dede7c622763640b5",
            "address":  "185.150.160.209:9735",
            "bytes_sent":  "367",
            "bytes_recv":  "367",
            "sat_sent":  "0",
            "sat_recv":  "0",
            "inbound":  false,
            "ping_time":  "106158",
            "sync_type":  "PASSIVE_SYNC",
            "features":  {
                "0":  {
                    "name":  "data-loss-protect",
                    "is_required":  true,
                    "is_known":  true
                },
                "5":  {
                    "name":  "upfront-shutdown-script",
                    "is_required":  false,
                    "is_known":  true
                },
                "7":  {
                    "name":  "gossip-queries",
                    "is_required":  false,
                    "is_known":  true
                },
                "9":  {
                    "name":  "tlv-onion",
                    "is_required":  false,
                    "is_known":  true
                },
                "12":  {
                    "name":  "static-remote-key",
                    "is_required":  true,
                    "is_known":  true
                },
                "14":  {
                    "name":  "payment-addr",
                    "is_required":  true,
                    "is_known":  true
                },
                "17":  {
                    "name":  "multi-path-payments",
                    "is_required":  false,
                    "is_known":  true
                },
                "23":  {
                    "name":  "anchors-zero-fee-htlc-tx",
                    "is_required":  false,
                    "is_known":  true
                },
                "27":  {
                    "name":  "shutdown-any-segwit",
                    "is_required":  false,
                    "is_known":  true
                },
                "31":  {
                    "name":  "amp",
                    "is_required":  false,
                    "is_known":  true
                },
                "45":  {
                    "name":  "explicit-commitment-type",
                    "is_required":  false,
                    "is_known":  true
                },
                "2023":  {
                    "name":  "script-enforced-lease",
                    "is_required":  false,
                    "is_known":  true
                }
            },
            "errors":  [],
            "flap_count":  1,
            "last_flap_ns":  "1714471490124671000",
            "last_ping_payload":  "00a0b521077b0c5673af50568eaad1a4e41f72983b982cc7845b010000000000000000004b2609c7cd78f46a2ab6a0f8536b0d709c44fe803380ca8a193c9bfb6ed64bc7a8bc3066db3103172ac6355e"
        },
        {
            "pub_key":  "031678745383bd273b4c3dbefc8ffbf4847d85c2f62d3407c0c980430b3257c403",
            "address":  "34.65.47.142:9735",
            "bytes_sent":  "367",
            "bytes_recv":  "367",
            "sat_sent":  "0",
            "sat_recv":  "0",
            "inbound":  false,
            "ping_time":  "132847",
            "sync_type":  "PASSIVE_SYNC",
            "features":  {
                "0":  {
                    "name":  "data-loss-protect",
                    "is_required":  true,
                    "is_known":  true
                },
                "5":  {
                    "name":  "upfront-shutdown-script",
                    "is_required":  false,
                    "is_known":  true
                },
                "7":  {
                    "name":  "gossip-queries",
                    "is_required":  false,
                    "is_known":  true
                },
                "9":  {
                    "name":  "tlv-onion",
                    "is_required":  false,
                    "is_known":  true
                },
                "12":  {
                    "name":  "static-remote-key",
                    "is_required":  true,
                    "is_known":  true
                },
                "14":  {
                    "name":  "payment-addr",
                    "is_required":  true,
                    "is_known":  true
                },
                "17":  {
                    "name":  "multi-path-payments",
                    "is_required":  false,
                    "is_known":  true
                },
                "19":  {
                    "name":  "wumbo-channels",
                    "is_required":  false,
                    "is_known":  true
                },
                "23":  {
                    "name":  "anchors-zero-fee-htlc-tx",
                    "is_required":  false,
                    "is_known":  true
                },
                "27":  {
                    "name":  "shutdown-any-segwit",
                    "is_required":  false,
                    "is_known":  true
                },
                "31":  {
                    "name":  "amp",
                    "is_required":  false,
                    "is_known":  true
                },
                "45":  {
                    "name":  "explicit-commitment-type",
                    "is_required":  false,
                    "is_known":  true
                },
                "2023":  {
                    "name":  "script-enforced-lease",
                    "is_required":  false,
                    "is_known":  true
                }
            },
            "errors":  [],
            "flap_count":  1,
            "last_flap_ns":  "1714471490152438000",
            "last_ping_payload":  "00a0b521077b0c5673af50568eaad1a4e41f72983b982cc7845b010000000000000000004b2609c7cd78f46a2ab6a0f8536b0d709c44fe803380ca8a193c9bfb6ed64bc7a8bc3066db3103172ac6355e"
        }
    ]
}

```

**Reply by saubyk** [reply]
Given that your node is reporting that `synced_to_chain` is true, the error you posted doesn't seem relevant to the issue you're experiencing. May be there are few bitcoin nodes you're connected with, which are not serving blocks but that shouldn't stop your node from syncing the graph with the LN peers

Couple of things to try here:
1. remove the problematic peers add other bitcoin peers (perhaps one of these: https://bitnodes.io/nodes/?q=1097), to address the LNWL warning you're getting in the log
2. check your network settings and if the node's lightning port (default: 9735) is accessible or not from outside
3. post bitcoin.conf and lnd.conf (redact any sensitive info) for further troubleshooting

**Reply by ghost** [reply]
My bitcoin.conf file:


```
server=1
daemon=1 
txindex=0
disablewallet=0

rpcuser=helloworld
rpcpassword=helloworld123
rpcbind=127.0.0.1   
rpcallowip=127.0.0.1
rpcport=8332

dbcache=1000  
maxmempool=300

listen=1
maxconnections=40

# ZMQ options
zmqpubrawblock=tcp://127.0.0.1:28332
zmqpubrawtx=tcp://127.0.0.1:28333


prune=20000
```



My lnd.conf file:

```

debuglevel=info
maxpendingchannels=5
alias=YourNodeAlias
color=#68F442


bitcoin.active=1
bitcoin.mainnet=1
bitcoin.node=bitcoind


bitcoind.rpchost=localhost
bitcoind.rpcuser=helloworld
bitcoind.rpcpass=helloworld123
bitcoind.zmqpubrawblock=tcp://127.0.0.1:28332
bitcoind.zmqpubrawtx=tcp://127.0.0.1:28333

```

**Reply by saubyk** [reply]
> check your network settings and if the node's lightning port (default: 9735) is accessible or not from outside

Any response to the above? is the lightning port accessible?

**Reply by ghost** [reply]
I forgot to mention that I'm running a VPN that doesn't allow port forwarding.

I went on https://www.yougetsignal.com/tools/open-ports/ and it says that Port 9735 is closed. Is that a good way to check?

**Reply by ghost** [reply]
@saubyk

**Reply by saubyk** [reply]
Sorry this slipped off my radar. My prognosis so far:
- the warning you're getting in the log doesn't seem to be relevant to the issue. Did you try removing the problematic Bitcoin peers? If the nodes causing warning messages are removed, we should be able view actual error messages in the log
- the problem does seem to be the network connectivity of your port, but if you do have some n/w peers, you should get graph updates even with the outgoing connections
- try manually connecting to some of the big network peers. You can find them on https://1ml.com/
- update `debuglevel` to `debug` so that we can observe more information in the log
- also I am assuming you've restarted the node a few times

**Reply by ghost** [reply]
@saubyk  I just deleted my peers.dat bitcoin core file to get rid of problematic Bitcoin peers, and changed ```debuglevel``` to ```debug```.

It's still saying "synced_to_graph = false".

Also, by manually connecting to some big network peers, do you mean I should try opening a channel with them and send funds?

In any case, here's a screenshot of the terminal log. I'm getting a lot more messages after enabling ```debug```

<img width="1612" alt="new" src="https://github.com/lightningnetwork/lnd/assets/155139404/18681734-bcbe-431e-ac94-8710b3b2b510">

**Reply by saubyk** [reply]
> Also, by manually connecting to some big network peers, do you mean I should try opening a channel with them and send funds?

I meant, just adding more network peers so that your node gets data from other network graphs. Not opening channel or anything else.

In any case from the log it appears that the node is getting graph updates just fine. How long have you been trying to sync the graph? Are you keeping the node online all the time?

**Reply by ghost** [reply]
@saubyk I've tried keeping it on for hours on end, closing/reopening dozens of times and nothing changes. It's running on my laptop, so it can't be on 24/7, only when I use the computer I can have it running. Usually it's for a few hours before I close it.

How do I add more network peers ? Could you let me know the exact terminal command I should type?

**Reply by saubyk** [reply, marked as answer]
Here's the command:
```
$ lncli connect --help

NAME:
   lncli connect - Connect to a remote lightning peer.

USAGE:
   lncli connect [command options] <pubkey>@host
```
More details here: https://lightning.engineering/api-docs/api/lnd/lightning/connect-peer

Also please note that it's critical for lightning nodes to be **always online**. I would not recommend running node on your work laptop which is periodically switched off. You need to get a dedicate device to run a lightning node, which must be on and online all the time.

**Reply by ghost** [reply]
@saubyk Thanks. I ended up being able to connect to a node, opened a channel with them for arround 500k sats, but still unable to send any transactions.

Everytime I attempt a transaction, I get this error:  "Payment status: FAILED, reason: FAILURE_REASON_INSUFFICIENT_BALANCE"

Even though I'm only trying to make a transaction for 5000 satoshis, and the channel size I opened was 500,000 satoshis. The channel I opened was with ACINQ.

**Reply by saubyk** [reply]
Is the channel opening transaction confirmed and the channel status active?

**Reply by ghost** [reply]
@saubyk Yes, that's correct, the channel opening transaction was confirmed. I tried about 10 transactions since yesterday, and they all failed with the same error message.

I noticed here however that it says ```"active": false```, so perhaps the channel status is inactive? Not sure why.

Here's what my terminal is logging when I type ```lncli listpeers```:

```
 [
        {
            "active":  false,
            "remote_pubkey":  "censored",
            "channel_point":  “censored:1”,
            "chan_id":  "censored",
            "capacity":  "censored",
            "local_balance":  "censored",
            "remote_balance":  "0",
            "commit_fee":  "3140",
            "commit_weight":  "772",
            "fee_per_kw":  "2500",
            "unsettled_balance":  "0",
            "total_satoshis_sent":  "0",
            "total_satoshis_received":  "0",
            "num_updates":  "0",
            "pending_htlcs":  [],
            "csv_delay":  720,
            "private":  false,
            "initiator":  true,
            "chan_status_flags":  "ChanStatusDefault",
            "local_chan_reserve_sat":  "4000",
            "remote_chan_reserve_sat":  "4000",
            "static_remote_key":  false,
            "commitment_type":  "ANCHORS",
            "lifetime":  "1690",
            "uptime":  "1448",
            "close_address":  "",
            "push_amount_sat":  "0",
            "thaw_height":  0,
            "local_constraints":  {
                "csv_delay":  720,
                "chan_reserve_sat":  "4000",
                "dust_limit_sat":  "354",
                "max_pending_amt_msat":  "180000000",
                "min_htlc_msat":  "1",
                "max_accepted_htlcs":  30
            },
            "remote_constraints":  {
                "csv_delay":  144,
                "chan_reserve_sat":  "4000",
                "dust_limit_sat":  "546",
                "max_pending_amt_msat":  "396000000",
                "min_htlc_msat":  "1",
                "max_accepted_htlcs":  483
            },
            "alias_scids":  [],
            "zero_conf":  false,
            "zero_conf_confirmed_scid":  "0",
            "peer_alias":  "ACINQ",
            "peer_scid_alias":  "0",
            "memo":  ""
        }
    ]
}
```

the "censored" was added by me to protect my privacy

**Reply by saubyk** [reply]
If the channel opening transaction was confirmed and is more than 6 blocks deep, the channel status should be `"active" : true`. If for whatever reason the channel is not active (which seems to be the case for you), the transactions will not succeed.

- Is your node synced to chain and graph?
- What's the output of `lncli pendingchannels` and `lncli listchannels`?
- Can you share the channel point, so that we can inspect the transaction onchain?

**Reply by ghost** [reply]
@saubyk Thanks for your quick response, it's extremely appreciated. I was able to finally make my first ever successful transaction on the lightning network by closing LND & bitcoin core, and reopening them!

Since then, I've made around 6-7 successful transactions, however I noticed that some payments required me to try multiple times before it worked. 

The last payment I attempted was stuck on "IN FLIGHT" for about a minute, then said something along the lines of "failed to find route". This happened at least 3 times, then it worked. Then I tried again and the same issue occured. Payment error "failed to find route".

Keep in mind that i'm still not synced to the graph while all of this is happening.

What could be causing this?

**Reply by saubyk** [reply]
There can be multiple explanations to what your experience is:
- given that your node is not synced to the graph, it may not have complete picture of the network and hence is selecting sub-optimal paths to make the payment
- it also depends on the destination that you're trying to send the payment to, and how well connected it is with the network or the liquidity situation on its channels

Regardless as advised earlier, you must run your node on a dedicated device which is online all the time and is synced with chain and graph. Lightning nodes are not meant to run on devices which are shutdown frequently

**Reply by bitromortac** [reply]
Could you report the output of `lncli getnetworkinfo`, I'd be interested in your general network view @drkhero.

**Reply by ghost** [reply]
@bitromortac @saubyk As requested here is the output of ```lncli getnetworkinfo```:

```
{
    "graph_diameter":  13,
    "avg_out_degree":  3.971243042671614,
    "max_out_degree":  91,
    "num_nodes":  15092,
    "num_channels":  29967,
    "total_network_capacity":  "96884431081",
    "avg_channel_size":  3233037.3771481966,
    "min_channel_size":  "1050",
    "max_channel_size":  "640000000",
    "median_channel_size_sat":  "860879",
    "num_zombie_chans":  "19166"
}


```

I'm still not synced to the graph, still have no idea why it refuses to let me be synced to the graph. Perhaps a bug, or something deeper? I have no idea.

**Comment by saubyk** [comment]
Also moving this to a discussion, as it's a troubleshooting request
